pkg/creational/pool/pool.go

```diff
@@ -145,19 +145,16 @@ func (rpm *ResourcePoolManager[T]) AcquireResource(ctx context.Context, isNeedTo
 // Acquisition of resources is thread-safe.
 func (rpm *ResourcePoolManager[T]) getResource(resourceObtained chan<- resourceObtainer[T]) {
 	var acqManagedResource *managedResource[T]
+
 	rpm.pool.Range(func(_, value any) bool {
 		mr, ok := value.(*managedResource[T])
 		if !ok {
-			return false
+			return true
```

Comment:
have a feeling that this could also be a problematic interaction, since we would want to loop anyway through all the resources?